### PR TITLE
[Windwalker] Implementing Checklist

### PR DESCRIPTION
**Pull request by Juko8**
Example log: 
https://www.warcraftlogs.com/reports/NJ9TBxdWRFLfQKyq#fight=48&type=damage-done
billede
billede

### CODE REVIEW

File: src/Parser/Monk/Windwalker/Modules/Features/Checklist.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.INVOKE_XUEN_THE_WHITE_TIGER_TALENT,
+            when: combatant.hasTalent(SPELLS.INVOKE_XUEN_THE_WHITE_TIGER_TALENT.id),
+          }),
+        ];
+      },
+    }),
+    new Rule({
+      name: 'Manage your resources',
+      description: <Wrapper>Windwalker is heavily dependent on having enough Chi to cast your core spells on cooldown. Wasting Chi either by generating while capped or using <SpellLink id={SPELLS.BLACKOUT_KICK.id}icon/> and <SpellLink id={SPELLS.SPINNING_CRANE_KICK.id}icon/> too much will cause you to delay your hard hitting Chi spenders and lose damage.</Wrapper>,
```
Comment (Gebuz) [inline src/Parser/Monk/Windwalker/Modules/Features/Checklist.js]: You mention using blackout kick too much is bad but none of the metrics track that. Did you forget to add it or is there just no module for tracking that?

File: src/Parser/Monk/Windwalker/Modules/Spells/ComboStrikes.js [hunk 2]
```diff
@@ -50,14 +50,24 @@ class ComboStrikes extends Analyzer {
     this._lastSpellUsed = spellId;
   }
 
+  get suggestionThresholds() {
+    return {
+      actual: this.masteryDropSpellSequence.length,
```
[the remark this answers is not in the document]
Comment (Juko8) [reply src/Parser/Monk/Windwalker/Modules/Spells/ComboStrikes.js]: I'm not really a fan of telling people that they dropped mastery 0.3 times/minute and that was a mistake. The same goes for SpinningCraneKick.badCasts.

File: src/Parser/Monk/Windwalker/Modules/Features/Checklist.js (hunk 1 above)
Comment (Juko8) [reply src/Parser/Monk/Windwalker/Modules/Features/Checklist.js]: There is no metric, it's only really measurable by having low cast efficiency on cooldown Chi spenders. I have thought of looking at a "Chi spent on Blackout Kick vs other chi spenders ratio" but it's going to be tricky finding the correct ratio to measure by.

Comment (Gebuz) [reply src/Parser/Monk/Windwalker/Modules/Features/Checklist.js]: maybe you could do something like Blackout kicks cast while other spenders were available? Anyway you don't to add it here just a thing for the future.

File: src/Parser/Monk/Windwalker/Modules/Spells/ComboStrikes.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gebuz) [reply src/Parser/Monk/Windwalker/Modules/Spells/ComboStrikes.js]: that's up to you, just seems like a pretty rough threshold on long encounters that it is a major mistake if you mess it up once.

**Pull request merged**